src/python/WMCore/WMSpec/StdSpecs/PromptReco.py

```diff
@@ -61,7 +61,7 @@ def buildWorkload(self):
                                                splitAlgo = self.procJobSplitAlgo,
                                                splitArgs = self.procJobSplitArgs,
                                                stepType = cmsswStepType,
-                                               forceUnmerged = True)
+                                               forceUnmerged = ["write_ALCARECO"] if 'ALCARECO' in self.writeTiers else False)
```

Comment:
Also, two parameters makes for ugly code as well when you have to make a decision based on them (do this based on parameter A but only if parameter B isn't set and has this value).